tests/test_tuners_utils.py

```diff
@@ -175,3 +231,80 @@ def test_feedforward_matching_ia3(self):
                 self.assertTrue(module.is_feedforward)
             else:  # other IA3 modules should not be marked as feedforward
                 self.assertFalse(module.is_feedforward)
+
+    @parameterized.expand(MAYBE_INCLUDE_ALL_LINEAR_LAYERS_TEST_CASES)
+    def test_maybe_include_all_linear_layers_lora(
+        self, model_id, model_type, initial_target_modules, expected_target_modules
+    ):
+        model = self.transformers_class_map[model_type].from_pretrained(model_id)
+        config_cls = LoraConfig
+        self._check_match_with_expected_target_modules(
+            model_id, model, config_cls, initial_target_modules, expected_target_modules
+        )
+
+    @parameterized.expand(BNB_TEST_CASES)
```

Comment:
Nit: Let's move the `@parameterized.expand` to be the innermost decorator.

Reply:
This requires some more work! Currently, what's happening is that the `parameterized.expand` creates multiple test cases with the given arguments, and then the GPU and BNB checks run for each test case. This is fine but a bit wasteful. However, if I add `parameterized.expand` to the end, then the test case isn't skipped! I believe there's some weird interaction between `unittest.skip`/`pytest.mark.skip` and `paramterized` because of which the test case doesn't get skipped (locally, I see errors because I'm using a Macbook). I'll need to look into this a bit more and will make another PR with a fix.

Reply:
Weird. Never mind then, it's not super important, that's why I called it a "nit".